src/main/scala/io/iohk/ethereum/signer/EIP2930Signer.scala

```diff
@@ -0,0 +1,70 @@
+package io.iohk.ethereum.signer
+
+import akka.util.ByteString
+
+import io.iohk.ethereum.crypto
+import io.iohk.ethereum.crypto.ECDSASignature
+import io.iohk.ethereum.domain.LegacyTransaction
+import io.iohk.ethereum.domain.SignedTransaction
+import io.iohk.ethereum.domain.Transaction
+import io.iohk.ethereum.domain.TransactionWithAccessList
+import io.iohk.ethereum.network.p2p.messages.BaseETH6XMessages.accessListItemCodec
+import io.iohk.ethereum.rlp.RLPImplicitConversions._
+import io.iohk.ethereum.rlp.RLPImplicits._
+import io.iohk.ethereum.rlp.RLPList
+import io.iohk.ethereum.rlp.{encode => rlpEncode, _}
+
+/** https://eips.ethereum.org/EIPS/eip-2930
+  * EIP 2930 introduces the new Transaction Type 1, with access list
+  */
+class EIP2930Signer(val chainId: BigInt) extends Signer {
+
+  private val eip155Signer = new EIP155Signer(chainId)
+
+  override def signatureFromBytes(bytes65: ByteString): Option[ECDSASignature] = ???
+
+  override def payloadToCheck(signedTransaction: SignedTransaction): Either[SignerError, Array[Byte]] =
+    // actually, payload to check is not required to include transaction type
```

Comment:
I understand the "SHOULD" as : "some transaction types in the future might define a signature mechanism which does not include the type in the data", not as "the sender has a choice on how he can sign the transaction".

As EIP-2930 for transaction type 1 clearly states "Signature signs over the transaction type as well as the transaction data", I think there is no ambiguity there.

Reply:
Effectively, I had misunderstood the EIP-2718 part where it's specificed that the signature SHOULD but MUST NOT sign over the transaction part.
On second read, it just let the possibillity for each transaction type specification to choose its signature content.

Thanks for pointing this.